src/lib/services/LocalCatalog.ts

```diff
@@ -146,12 +146,21 @@ export default class LocalCatalog implements RuleCatalog {
 	}
 
 	private async rebuildCatalog(): Promise<[boolean, string]> {
-		this.catalog = {rulesets:[], categories:[], rules: []};
-		for(const engine of this.engines) {
-			const catalog: Catalog = await engine.getCatalog();
-			catalog.rulesets.forEach(rs => {rs.engine = engine.getName(); this.catalog.rulesets.push(rs);});
-			catalog.categories.forEach(c => {c.engine = engine.getName(); this.catalog.categories.push(c);});
-			catalog.rules.forEach(r => {r.engine = engine.getName(); this.catalog.rules.push(r);});
+		const catalog = this.catalog = {rulesets: [], categories: [], rules: []};
```

Comment:
This whole method is begging for some commenting.

Reply:
On second thought, this was lame.  I went back and changed PmdEngine to set the engine itself when creating its catalog.  This part here goes back to being trivial.